File: #1021_removeOuterParentheses.py
class Solution:
    def removeOuterParentheses(self, S: str) -> str:
        stack = []
        opend = 0
        for i in S:
            if i == '(' and opend>0:
                stack.append(i)
            if i == ')' and opend>1:
                stack.append(i)
            opend+=1 if i == '(' else -1
        return ''.join(stack)

File: Java/#1021_removeOuterParentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        String result="";
        int open=0;
        String temp="";
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='('){
                temp+=Character.toString(s.charAt(i));
                open++;
            }else{
                temp+=Character.toString(s.charAt(i));
                open--;
            }
            if(open==0){
                result+=temp.substring(1,temp.length()-1);
                temp="";
            }
        }
        return result;
    }
}
